### PR TITLE
Bug 1871737: Update status using subresource /status

### PR DESCRIPTION
**Pull request by periklis**
### Description
This PR addresses a prerequisite to enable CRD v1 generation. In detail, it moves updates to the `Elasticsearch` CR for status to the status subresource. Furthermore this practice conforms with k8s [general api conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status)

/cc @ewolinetz @blockloop

### CODE REVIEW

File: bundle/manifests/logging.openshift.io_kibanas_crd.yaml [hunk 1]
```diff
@@ -156,7 +156,55 @@ spec:
...
+                        description: Last time the condition transitioned from one
+                          status to another.
+                        format: date-time
+                        type: string
+                      message:
+                        description: Human-readable message indicating details about
+                          last transition.
+                        type: string
+                      reason:
+                        description: Unique, one-word, CamelCase reason for the condition's
```
Comment (blockloop) [inline bundle/manifests/logging.openshift.io_kibanas_crd.yaml:176]: nit: ~CamelCase~ PascalCase

Comment (periklis) [reply bundle/manifests/logging.openshift.io_kibanas_crd.yaml:176]: No chance to change this wording. It comes from deps.

**Pull request merged**